src/middlewared/middlewared/plugins/nfs.py

```diff
@@ -84,13 +85,27 @@ async def nfs_extend(self, nfs):
         nfs["userd_manage_gids"] = nfs.pop("16")
         nfs["v4_owner_major"] = nfs.pop("v4_owner_major")
         nfs["keytab_has_nfs_spn"] = keytab_has_nfs
+
+        # 'None' indicates we are to dynamically manage the number of nfsd
+        if nfs['servers'] is None:
+            nfs['managed_nfsd'] = True
+            cpu_info = await self.middleware.call("system.cpu_info")
+
+            # Default calculation:
+            #     Number of nfsd == number of cores, but not zero or greater than 32
+            nfs['servers'] = min(max(cpu_info['core_count'], 1), 32)
```

Comment:
If we have `servers` set to `null` and you update any other attribute in the NFS configuration, you will set `servers` value automatically which won't be desired behavior as user is not setting it

Reply:
Good catch!  You are correct!